glean-core/src/metrics/datetime.rs

```diff
@@ -0,0 +1,119 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::metrics::time_unit::TimeUnit;
+use crate::metrics::Metric;
+use crate::metrics::MetricType;
+use crate::storage::StorageManager;
+use crate::util::get_iso_time_string;
+use crate::CommonMetricData;
+use crate::Glean;
+
+use chrono::{DateTime, FixedOffset, TimeZone};
+
+#[derive(Debug)]
+pub struct DatetimeMetric {
+    meta: CommonMetricData,
+    time_unit: TimeUnit,
+}
+
+impl MetricType for DatetimeMetric {
+    fn with_meta(meta: CommonMetricData) -> Self {
+        Self {
+            meta,
+            time_unit: TimeUnit::Day,
+        } // FIXME: How do we handle this?
```

Comment:
I'm not sure how to handle this case. This is relevant for the labelled metrics. @mdboom @badboy , thoughts?

Reply:
Yeah, so I knew about this shortcoming when I implemented this for labeled metrics.

We essentially have 2 options I think (didn't test them yet):

1. We move `time_unit` into common metric data as an optional field (and then have to do that with multiple of the other optional fields)
2. We keep around the data and in labeled metrics we pass in a constructed instance instead of just keeping the type around (has additional memory overhead)

Reply:
> 
>     2. We keep around the data and in labeled metrics we pass in a constructed instance instead of just keeping the type around (has additional memory overhead)

I think I would favour this, but I'm not quite sure what do you mean exactly by it. Can you expand a bit?

Reply:
Yeah, so this is the change: https://github.com/mozilla/glean/commit/46e490835784fba1459ed751a14f454828b6767e

(It's independent from your part, so we could easily land that before/after)

Reply:
Thanks, I've rebased on #114 !